js/main.js

```diff
@@ -44,7 +44,8 @@ $form.addEventListener('submit', function (event) {
 
     for (let i = 0; i < $list.childNodes.length; i++) {
       if ($list.childNodes[i].tagName === 'LI' && $list.childNodes[i].dataset.entryId * 1 === data.editing.entryId) {
-        $list.childNodes[i].replaceWith(renderEntry(data.editing));
+        $list.childNodes[i].replaceWith(renderEntry(messageData));
+        $list.childNodes[i].setAttribute('data-entry-id', data.editing.entryId);
```

Comment:
You're already setting the attribute inside of `renderEntry`, so no need to do this again